content-review/impl/compilatio/src/main/java/org/sakaiproject/contentreview/compilatio/CompilatioReviewServiceImpl.java

```diff
@@ -0,0 +1,1084 @@
+			
+			try {
+				//check if current item has to be processed after the assignment due date
+				String assignmentId = assignmentService.getEntity(entityManager.newReference(currentItem.getTaskId())).getId();
+				Assignment a = assignmentService.getAssignment(assignmentId);
+				AssignmentContent ac = a.getContent();
+				
+				if(ac.getGenerateOriginalityReport().equals(NEW_ASSIGNMENT_REVIEW_SERVICE_REPORT_DUE)) {
+					Date dueDate = new Date(a.getDueTime().getTime());
+					if(dueDate.after(new Date())) {
+						log.debug("assignment due time not yet reached for item: " + currentItem.getId());
+						currentItem.setNextRetryTime(this.getNextRetryTime(0));
+						crqs.update(currentItem);
+						continue;
+					}
+				}
+			} catch (IdUnusedException | PermissionException e) {
+				log.error("Error getting assignment for item "+currentItem.getId(), e);
+				processError(currentItem, ContentReviewConstants.CONTENT_REVIEW_SUBMISSION_ERROR_NO_RETRY_CODE, null, null);
+				errors++;
+				continue;
+			}
+
+			log.debug("Attempting to submit content (status:"+currentItem.getStatus()+"): " + currentItem.getContentId() + " for user: "
+					+ currentItem.getUserId() + " and site: " + currentItem.getSiteId());						
+
+			if (currentItem.getRetryCount() == null) {
+				currentItem.setRetryCount(Long.valueOf(0));
+				currentItem.setNextRetryTime(this.getNextRetryTime(0));
+				crqs.update(currentItem);
+			} else if (currentItem.getRetryCount().intValue() > maxRetry) {
+				processError(currentItem, ContentReviewConstants.CONTENT_REVIEW_SUBMISSION_ERROR_RETRY_EXCEEDED_CODE, null, null);
+				errors++;
+				continue;
+			} else {
+				long l = currentItem.getRetryCount().longValue();
+				l++;
+				currentItem.setRetryCount(Long.valueOf(l));
+				currentItem.setNextRetryTime(this.getNextRetryTime(Long.valueOf(l)));
+				crqs.update(currentItem);
+			}
+			
+			//if document has no external id, we need to add it to compilatio
+			if(StringUtils.isBlank(currentItem.getExternalId())) {
+				//check if we have added it correctly
+				if(addDocumentToCompilatio(currentItem) == false){
+					errors++;
+					continue;
+				}
+			}
+			
+			Document document = null;
+			// Start Compilation Analyse
+			try {
+				Map params = CompilatioAPIUtil.packMap("action", "startDocumentAnalyse", "idDocument", currentItem.getExternalId());
+
+				document = compilatioConn.callCompilatioReturnDocument(params);
+
+			} catch (TransientSubmissionException e) {
+				processError(currentItem, ContentReviewConstants.CONTENT_REVIEW_SUBMISSION_ERROR_RETRY_CODE, "Error Submitting Assignment for Submission: " + e.getMessage() + ". Assume unsuccessful", null);
+				errors++;
+				continue;
+			} catch (SubmissionException e) {
+				processError(currentItem, ContentReviewConstants.CONTENT_REVIEW_SUBMISSION_ERROR_RETRY_CODE, "Error Submitting Assignment for Submission: " + e.getMessage() + ". Assume unsuccessful", null);
+				errors++;
+				continue;
```

Comment:
You're doing the same thing in both of these catches, whynot combine them into one multi-catch?